src/expansion/path_exp.c

```diff
@@ -0,0 +1,55 @@
+/* ************************************************************************** */
+/*                                                                            */
+/*                                                        :::      ::::::::   */
+/*   path_exp.c                                         :+:      :+:    :+:   */
+/*                                                    +:+ +:+         +:+     */
+/*   By: fratajcz <marvin@42.fr>                    +#+  +:+       +#+        */
+/*                                                +#+#+#+#+#+   +#+           */
+/*   Created: 2020/02/13 14:14:32 by fratajcz          #+#    #+#             */
+/*   Updated: 2020/02/13 15:05:14 by fratajcz         ###   ########.fr       */
+/*                                                                            */
+/* ************************************************************************** */
+
+#include "shell.h"
+
+bool	is_match(char *str, char *pat, char quote)
+{
+	if (*str && (quote_start(str, 0, &quote) || quote_stop(str, 0, &quote)))
+		return (is_match(str + 1, pat, quote));
+	if (*pat == '*' && quote == NONE)
+		return (*str == '\0'
+			? is_match(str, pat + 1, quote)
+			: is_match(str, pat + 1, quote) || is_match(str + 1, pat, quote));
+	if (*str == '\0' && *pat == '\0')
+		return (true);
+	if (*str && *str == *pat)
+		return (is_match(str + 1, pat + 1, quote));
+	return (false);
+}
+
+bool	is_pattern(char *str)
+{
+	char	quote_status;
+	bool	bracket_open;
+	int		i;
+
+	quote_status = NONE;
+	i = -1;
+	while (str[++i])
```

Comment:
Is `is_pattern()` assumes to receive always a non-zero `str`?
Otherwise `str` should be verified before processing to avoid potential segmentation fault.

Reply:
It will segfault just like puts(NULL) would segfault, if a pointer has a chance to be NULL it should be checked in the function that calls is_match (in our case, only strings from t_dstr structs are passed to this function and they can't be null since they are always malloced with ft_dstr_new).

Reply:
https://softwareengineering.stackexchange.com/questions/186036/when-should-pointers-be-checked-for-null-in-c

Reply:
Got it!